tests/test_encoding.py

```diff
@@ -88,6 +89,300 @@ def test_encode_coded_const_infer_order(self) -> None:
+            base_data_type=DataType.A_INT32,
+            base_type_encoding=Encoding.TWOC,
+            is_highlow_byte_order=True,
+            used_mask=None)
+        self.assertEqual(encode_state.coded_message, bytes([0x00, 0x12, 0x34]))
+
+        decode_state = DecodeState(encode_state.coded_message)
+        decoded = decode_state.extract_atomic_value(
+            bit_length=24,
+            base_data_type=DataType.A_INT32,
+            base_type_encoding=Encoding.TWOC,
+            is_highlow_byte_order=True)
+        self.assertEqual(decoded, 0x1234)
+
+        encode_state = EncodeState()
+        encode_state.emplace_atomic_value(
+            internal_value=-0x1234,
+            bit_length=24,
+            base_data_type=DataType.A_INT32,
+            base_type_encoding=Encoding.TWOC,
+            is_highlow_byte_order=True,
+            used_mask=None)
+        self.assertEqual(encode_state.coded_message, bytes([0xff, 0xed, 0xcc]))
+
+        decode_state = DecodeState(encode_state.coded_message)
+        decoded = decode_state.extract_atomic_value(
+            bit_length=24,
+            base_data_type=DataType.A_INT32,
+            base_type_encoding=Encoding.TWOC,
+            is_highlow_byte_order=True)
+        self.assertEqual(decoded, -0x1234)
+
+        # sign-magnitude
+        encode_state = EncodeState()
+        encode_state.emplace_atomic_value(
+            internal_value=0x1234,
+            bit_length=24,
+            base_data_type=DataType.A_INT32,
+            base_type_encoding=Encoding.SM,
+            is_highlow_byte_order=True,
+            used_mask=None)
+        self.assertEqual(encode_state.coded_message, bytes([0x00, 0x12, 0x34]))
+
+        decode_state = DecodeState(encode_state.coded_message)
+        decoded = decode_state.extract_atomic_value(
+            bit_length=24,
+            base_data_type=DataType.A_INT32,
+            base_type_encoding=Encoding.SM,
+            is_highlow_byte_order=True)
+        self.assertEqual(decoded, 0x1234)
+
+        encode_state = EncodeState()
+        encode_state.emplace_atomic_value(
+            internal_value=-0x1234,
+            bit_length=24,
+            base_data_type=DataType.A_INT32,
+            base_type_encoding=Encoding.SM,
+            is_highlow_byte_order=True,
+            used_mask=None)
+        self.assertEqual(encode_state.coded_message, bytes([0x80, 0x12, 0x34]))
+
+        decode_state = DecodeState(encode_state.coded_message)
+        decoded = decode_state.extract_atomic_value(
+            bit_length=24,
+            base_data_type=DataType.A_INT32,
+            base_type_encoding=Encoding.SM,
+            is_highlow_byte_order=True)
+        self.assertEqual(decoded, -0x1234)
+
+    def test_float_encodings(self) -> None:
+        # FLOAT32
+        encode_state = EncodeState()
+        with self.assertRaises(OdxError):
+            encode_state.emplace_atomic_value(
+                internal_value=-1.234,
```

Comment:
Add test for magic float values, like NaN and infinity

Reply:
done: https://github.com/mercedes-benz/odxtools/pull/381/commits/6a6a37d7380c31eca698d3bafde4a1362dc63818